internal/printer/printer_test.go

```diff
@@ -811,6 +811,20 @@ const groups = [[0, 1, 2], [3, 4, 5]];
 				code: "${$$maybeRenderHead($$result)}<body><!-- \\`npm install astro\\` --></body>",
 			},
 		},
+		{
+			name:   "HTML comment in component inside expression I",
+			source: "{(() => <Component><!--Hi--></Component>)}",
+			want: want{
+				code: "${(() => $$render`${$$renderComponent($$result,'Component',Component,{},{})}`)}",
```

Comment:
Oh that's true, I'll fix that

Reply:
Actually, it looks like comments should not be slotted, I found a [test case](https://github.com/withastro/compiler/blob/0bfdc2d0763bd79ccc44bd722bb19b8daec9b668/internal/printer/printer_test.go#L961) for that. Here's [another test](https://github.com/withastro/compiler/blob/0bfdc2d0763bd79ccc44bd722bb19b8daec9b668/internal/printer/printer_test.go#L809) showing comments being preserved outside component slots

Reply:
Ah you're totally correct, it would be strange to pass these comments into framework components. I didn't think about this being _inside_ of the `<Component>` node.